File: src/main/java/we/code/demo/model/entity/Address.java
package we.code.demo.model.entity;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class Address {
    public enum State {
        QLD,
        NSW,
        ACT,
        VIC,
        TAS,
        NT,
        SA,
        WA
    }

    private String streetLine1;
    private String streetLine2;
    private String suburb;
    private State state;
    private String postCode;

    public Address(String streetLine1, String streetLine2, String suburb, State state, String postCode) {
        this.streetLine1 = streetLine1;
        this.streetLine2 = streetLine2;
        this.suburb = suburb;
        this.state = state;
        this.postCode = postCode;
    }
}

File: IoTBay/Models/Entities/Address.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace IoTBay.Models.Entities;

public enum State
{
    ACT, NSW, NT, QLD, SA, TAS, VIC, WA
}

public partial class Address
{
    public int AddressId { get; set; }

    public string? StreetLine1 { get; set; }

    public string? StreetLine2 { get; set; }

    public string? Suburb { get; set; }

    [Column("state")]
    public State? State { get; set; }

    public string? Postcode { get; set; }

    public virtual ICollection<ShipmentMethod> ShipmentMethods { get; set; } = new List<ShipmentMethod>();

    public virtual ICollection<Supplier> Suppliers { get; set; } = new List<Supplier>();
}
